### DISCUSSION TITLE
Call `foo.pb.js` functions from `bar.pb.js`

### DISCUSSION THREAD

**Discussion by benstigsen** [pocketbase/pocketbase#5760] [Q&A] [answered]
Hello, I'm loving PocketBase so far! But I'm having a problem.

It's very likely I'm just inexperienced, but how do I "import" or call a function in `bar.pb.js` from `foo.pb.js`?.

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to reuse functions and variables outside of the handlers scope you can register them as CJS module (_the current runtime doesn't support ESM style of imports at the moment_). You can find more details about this and other limitations in https://pocketbase.io/docs/js-overview/#caveats-and-limitations.

For clarity, here is an example with a custom exported helper function:

```js
// pb_hooks/utils.js
module.exports = {
    hello: (name) => {
        console.log("Hello " + name)
    }
}
```

```js
// pb_hooks/main.pb.js

// PocketBase v0.23+
onBootstrap((e) => {
    e.next()

    const utils = require(`${__hooks}/utils.js`)

    utils.hello("world")
})


// <= PocketBase v0.22
onAfterBootstrap((e) => {
    const utils = require(`${__hooks}/utils.js`)

    utils.hello("world")
})
```

**Reply by benstigsen** [reply]
That's perfect, thank you very much!
I should've taken a better look at the documentation.

**Reply by synapse** [reply]
@ganigeorgiev  I am getting this error `ReferenceError: module is not defined` on `0.23.2`. Was this changed?

**Reply by ganigeorgiev** [reply]
@synapse There are no changes to the JS modules loading. The only thing that has changed in the above example is the `onAfterBootstrap` -> `onBootstrap` (_I've updated and tested the code sample above and it works correctly for me_).

**Reply by ganigeorgiev** [reply]
@synapse Make sure that the file with the module you are requiring doesn't have `.pb.js` extension (_otherwise the PocketBase loader JSVM will attempt to execute it_). It must be simply `.js`.

**Reply by synapse** [reply]
Yup @ganigeorgiev that was it ☝️ . Previously it didn't check for the `.pb.js` extension. Thanks for the help 👍
